src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/ASTAnalyser.java

```diff
@@ -237,6 +240,119 @@ private void visitClassOrInterfaceOrEnumDeclaration(TypeDeclaration<?> typeDecla
             addToken(new Token(NEW_LINE, ""));
         }
 
+        private void visitModuleDeclaration(ModuleDeclaration moduleDeclaration) {
+            addToken(new Token(KEYWORD, "module"));
+            addToken(new Token(WHITESPACE, " "));
+            addToken(new Token(TYPE_NAME, "module-info", "module-info"));
+            addToken(new Token(WHITESPACE, " "));
+            addToken(new Token(PUNCTUATION, "{"));
+            addToken(new Token(NEW_LINE, ""));
+
+            moduleDeclaration.getDirectives().forEach(moduleDirective -> {
+                indent();
+                addToken(makeWhitespace());
+
+                moduleDirective.ifModuleRequiresStmt(d -> {
+                    addToken(new Token(KEYWORD, "requires"));
+                    addToken(new Token(WHITESPACE, " "));
+
+                    if (d.isTransitive()) {
+                        addToken(new Token(KEYWORD, "transitive"));
+                        addToken(new Token(WHITESPACE, " "));
+                    }
+
+                    addToken(new Token(TYPE_NAME, d.getNameAsString()));
+                    addToken(new Token(PUNCTUATION, ";"));
+                    addToken(new Token(NEW_LINE, ""));
+                });
+
+                moduleDirective.ifModuleExportsStmt(d -> {
+                    addToken(new Token(KEYWORD, "exports"));
+                    addToken(new Token(WHITESPACE, " "));
+                    addToken(new Token(TYPE_NAME, d.getNameAsString()));
+
+                    NodeList<Name> names = d.getModuleNames();
+
+                    if (!names.isEmpty()) {
+                        addToken(new Token(WHITESPACE, " "));
+                        addToken(new Token(KEYWORD, "to"));
+                        addToken(new Token(WHITESPACE, " "));
+
+                        for (int i = 0; i < names.size(); i++) {
```

Comment:
since we use .size() more than once, it is better to create a final variable for it.

Reply:
yes, when perf is critical we should - but here it is ok as-is.